### PR TITLE
Added uploaderAdditionalFields to PXLPhoto

### PR DESCRIPTION
**Pull request by SungjunApp**
ticket: https://pixlee.atlassian.net/browse/IMP-629

# Overview
This is a request of Jones Soda. They want to upload content via our [API](https://developers.pixlee.com/reference#create-new-content-from-url) with `connected_user` to store their customizable JSON data. Then, our SDK needs to provide the field in PXLPhoto. 

FYI, this work's done on Android SDK.
# What's updated
- Added uploaderAdditionalFields to PXLPhoto that can accommodate dynamic and nested JSON

### CODE REVIEW

**Review by jchen123 — APPROVED** [review]

File: Classes/SDK/DTOs/PXLPhotoResponseDTO.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ struct PXLPhotoDTO: Codable {
     let scheduler: String?
     let engagementRate: Double?
     let markedAsSpam: Bool?
-    let uploaderAdditionalFields: PXLUploaderAdditionalFields?
+    let uploaderAdditionalFields: [String: Any]?
```
Comment (jharrilim) [inline Classes/SDK/DTOs/PXLPhotoResponseDTO.swift:57]: This field type is defined by the user of the SDK right? Perhaps we make this a generic that must implement the `Codable` protocol and let them decode it. That way we can skip trying to decode [Any], and they get a strongly typed interface

File: Classes/SDK/DTOs/PXLPhotoResponseDTO.swift [hunk 2]
```diff
@@ -17,7 +17,7 @@ struct PXLPhotoResponseDTO: Codable {
     }
 }
 
-struct PXLPhotoDTO: Codable {
+struct PXLPhotoDTO: Decodable {
```
Comment (jharrilim) [inline Classes/SDK/DTOs/PXLPhotoResponseDTO.swift:20]: eg.
```suggestion
struct PXLPhotoDTO<UploaderAdditionalFields: Decodable>: Decodable {
```

File: Classes/SDK/DTOs/PXLPhotoResponseDTO.swift (hunk 1 above)
Comment (jharrilim) [inline Classes/SDK/DTOs/PXLPhotoResponseDTO.swift:57]:
```suggestion
    let uploaderAdditionalFields: UploaderAdditionalFields?
```

**Comment by SungjunApp** [comment]
Thanks, @jharrilim for reviewing this. Providing a generic wrapper is very nice and lets us and the SDK users keep the codes clean get rid of the manual parser, `init(from decoder: Decoder) {..}`! 

So, I played with this around a bit and reached this conclusion: Due to the pros and cons below, I will keep using the codes of init(from decoder: Decoder) {..} until finding a solution that the existing not using `uploaderAdditionalFields` don't have to change their codes.

# For uploaderAdditionalFields
### Option 2: T - Generic wrapper 
#### Pros
- generic wrapper. the SDK users can use their own class for uploaderAdditionalFields.
- cleaner code. can remove 80 lines of the manual parser that is `init(from decoder: Decoder) {..}` of PXLPhotoDTO in PXLPhotoResponseDTO.swift
#### Cons
- If other existing SDK users who don't use `uploaderAdditionalFields` try to update the SDK to the latest version, they need to make any class for `uploaderAdditionalFields` and change their codes. Because the SDK codes, firing APIs from the SDK will require the app-side codes to pass the generic type.

### Option [String: Any] - manual parsing
#### Pros
- no code changes required to existing SDK users
- Since [String: Any] allows to accommodate any type[primitive types, json object, json array] and different json structures for each photo, the app-side can implement whatever they want to do.
   ```swift
   #!swift
   if let album = album {
            _ = PXLClient.sharedClient.loadNextPageOfPhotosForAlbum(album: album) { photos, _ in
                if let photos = photos {
                    self.pxlGridView.items = photos

                    // demonstrating the customized json uploaded by API
                    for photo in photos {
                        debugPrint("userAdditional: \(photo.uploaderAdditionalFields)")
                        if let item = photo.uploaderAdditionalFields?["age"] as? Int {
                            debugPrint(" - age: \(item)")
                        }

                        if let item = photo.uploaderAdditionalFields?["points"] as? [Double] {
                            debugPrint(" - points: \(item)")
                        }

                        if let item = photo.uploaderAdditionalFields?["goods"] as? [String:Any] {
                            debugPrint(" - goods Json Array: \(item)")
                        }

                        if let item = photo.uploaderAdditionalFields?["goods"] as? [String] {
                            debugPrint(" - goods String Array: \(item)")
                        }
                    }
                }
            }
        }
   ```
   Console
   ```
    "userAdditional: Optional([\"goods\": [\"price\": \"150\", \"currency\": \"UDS\", \"title\": \"air pods 1gen\"], \"name\": \"Sample data\", \"points\": [10.0, 20.0, 35.0], \"age\": 73])"
    " - age: 73"
    " - points: [10.0, 20.0, 35.0]"
    " - goods Json Array: [\"price\": \"150\", \"currency\": \"UDS\", \"title\": \"air pods 1gen\"]"
    "userAdditional: Optional([\"email\": \"b@b.com\", \"age\": 73, \"name\": \"Donald Trump\", \"points\": [10.0, 20.0, 35.0]])"
    " - age: 73"
    " - points: [10.0, 20.0, 35.0]"
    " - goods String Array: [\"Macbook 2021\", \"air pods\", \"iPhone 12 Pro\"]"
   ```
- 
#### Cons
- 80 lines of the manual parser in PXLPhotoDTO. There's a possibility of adding a new variable and missing to add its parser.
- app-side should have a parser of `uploaderAdditionalFields` like the example above
   ```swift
   #!swift
                        if let item = photo.uploaderAdditionalFields?["age"] as? Int {
                            debugPrint(" - age: \(item)")
                        }

                        if let item = photo.uploaderAdditionalFields?["points"] as? [Double] {
                            debugPrint(" - points: \(item)")
                        }

                        if let item = photo.uploaderAdditionalFields?["goods"] as? [String:Any] {
                            debugPrint(" - goods Json Array: \(item)")
                        }

                        if let item = photo.uploaderAdditionalFields?["goods"] as? [String] {
                            debugPrint(" - goods String Array: \(item)")
                        }
   ```

**Pull request merged**